protocols/v2/channels-sv2/src/server/extended.rs

```diff
@@ -83,7 +84,7 @@ use tracing::debug;
 /// - the channel's job factory
 /// - the channel's chain tip
 #[derive(Debug)]
-pub struct ExtendedChannel<'a, J>
+pub struct ExtendedChannel<'a, J, P = NoPersistence>
```

Comment:
this syntax is only useful for `struct`s that support raw initialization without a constructor

for example, if this pseudocode was a valid usage pattern:
```rust
let extended_channel = ExtendedChannel {
   ... // lots of parameters but no `persistence` 
}
```

however, this `struct` is meant to always be initialized with a constructor, so `P = NoPersistence` serves no purpose

```suggestion
pub struct ExtendedChannel<'a, J, P>
```